cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -0,0 +1,193 @@
+import os
+import json
+import requests
+import base64
+import time as _time
+
+from datetime import datetime
+from enum import Enum, auto
+
+from cdci_data_analysis.analysis import email_helper
+from ..analysis.exceptions import RequestNotUnderstood
+from ..flask_app.templates import body_article_product_gallery
+
+
+class ContentType(Enum):
+    ARTICLE = auto()
+    DATA_PRODUCT = auto()
+    OBSERVATION = auto()
+    ASTROPHYSICAL_ENTITY = auto()
+
+
+def discover_mmoda_pg_token():
```

Comment:
this should come from dispatcher config

Reply:
yes, given what we agreed with the modification of the author of the content

Reply:
Not even because of that. 
Even before modification: this file you'd find locally has to belong to dispatcher.
Maybe there is some misunderstanding of how this is to be used: it will run in the dispatcher production, right?

Reply:
> Maybe there is some misunderstanding of how this is to be used: it will run in the dispatcher production, right?

no misunderstanding, it's clear